src/keylessBackup/KeylessBackupProgress.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { useEffect } from 'react'
+import { StyleSheet, Text, BackHandler, View } from 'react-native'
+import { useTranslation } from 'react-i18next'
+import { NativeStackScreenProps } from '@react-navigation/native-stack'
+import { StackParamList } from 'src/navigator/types'
+import { Screens } from 'src/navigator/Screens'
+import { useSelector } from 'react-redux'
+import { ensurePincode, navigate, navigateHome } from 'src/navigator/NavigationService'
+import { keylessBackupStatusSelector } from 'src/keylessBackup/selectors'
+import GreenLoadingSpinnerToCheck from 'src/icons/GreenLoadingSpinnerToCheck'
+import GreenLoadingSpinner from 'src/icons/GreenLoadingSpinner'
+import { KeylessBackupFlow, KeylessBackupStatus } from 'src/keylessBackup/types'
+import { KeylessBackupEvents } from 'src/analytics/Events'
+import fontStyles from 'src/styles/fonts'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { Spacing } from 'src/styles/styles'
+import Button, { BtnSizes, BtnTypes } from 'src/components/Button'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import RedLoadingSpinnerToInfo from 'src/icons/RedLoadingSpinnerToInfo'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'keylessBackup/KeylessBackupProgress'
+
+function KeylessBackupProgress({
+  route,
+}: NativeStackScreenProps<StackParamList, Screens.KeylessBackupProgress>) {
+  const keylessBackupStatus = useSelector(keylessBackupStatusSelector)
+
+  const { t } = useTranslation()
+
+  // Disable back button on Android
+  useEffect(() => {
+    const backPressListener = () => true
+    BackHandler.addEventListener('hardwareBackPress', backPressListener)
+    return () => BackHandler.removeEventListener('hardwareBackPress', backPressListener)
+  }, [])
+
+  // TODO(ACT-781): Implement Restore flow designs
+  if (route.params.keylessBackupFlow === KeylessBackupFlow.Restore) {
+    return <></>
+  }
+
+  const onPressContinue = () => {
+    ValoraAnalytics.track(KeylessBackupEvents.cab_progress_completed_continue)
+    navigateHome()
+  }
+
+  const onPressManual = async () => {
+    ValoraAnalytics.track(KeylessBackupEvents.cab_progress_failed_manual)
+    try {
+      const pinIsCorrect = await ensurePincode()
```

Comment:
why do we have to ensure pincode here?

Reply:
This follows the same pattern that the backup quiz/info section currently follows when accessed via the drawer navigator.

Reply:
ah ok, seems like the ensure pincode could be self contained over copying this in a bunch of places. Not directly in scope of this PR I guess